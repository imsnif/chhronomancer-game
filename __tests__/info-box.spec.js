import 'react-native'
import React from 'react'
import InfoBox from '../components/info-box'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

test('InfoBox (default props) => renders correctly', () => {
  const tree = renderer.create(
    <InfoBox />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('InfoBox (with title) => renders correctly', () => {
  const tree = renderer.create(
    <InfoBox title='foo' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('InfoBox (with image) => renders correctly', () => {
  const tree = renderer.create(
    <InfoBox image={require('../assets/placeholders/me.jpeg')} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('InfoBox (with title and image) => renders correctly', () => {
  const tree = renderer.create(
    <InfoBox title='foo' image={require('../assets/placeholders/me.jpeg')} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
