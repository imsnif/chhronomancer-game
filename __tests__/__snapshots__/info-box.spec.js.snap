// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox (default props) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00ff00",
      "borderColor": "#00ff00",
      "borderRadius": 0,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "column",
      "height": 100,
      "margin": 5,
      "marginBottom": 15,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={undefined}
      style={
        Object {
          "backgroundColor": "black",
          "borderRadius": 0,
          "flex": 1,
          "height": 21,
          "margin": 2,
          "marginBottom": 1,
          "width": 10,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "flex": 9,
          "fontFamily": "px_sans_nouveaux",
          "paddingRight": 30,
          "textAlign": "center",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "borderColor": "black",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 3,
        "flexDirection": "column",
        "margin": 1,
      }
    }
  />
</View>
`;

exports[`InfoBox (with image) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00ff00",
      "borderColor": "#00ff00",
      "borderRadius": 0,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "column",
      "height": 100,
      "margin": 5,
      "marginBottom": 15,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "backgroundColor": "black",
          "borderRadius": 0,
          "flex": 1,
          "height": 21,
          "margin": 2,
          "marginBottom": 1,
          "width": 10,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "flex": 9,
          "fontFamily": "px_sans_nouveaux",
          "paddingRight": 30,
          "textAlign": "center",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "borderColor": "black",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 3,
        "flexDirection": "column",
        "margin": 1,
      }
    }
  />
</View>
`;

exports[`InfoBox (with title and image) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00ff00",
      "borderColor": "#00ff00",
      "borderRadius": 0,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "column",
      "height": 100,
      "margin": 5,
      "marginBottom": 15,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "backgroundColor": "black",
          "borderRadius": 0,
          "flex": 1,
          "height": 21,
          "margin": 2,
          "marginBottom": 1,
          "width": 10,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "flex": 9,
          "fontFamily": "px_sans_nouveaux",
          "paddingRight": 30,
          "textAlign": "center",
        }
      }
    >
      foo
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "borderColor": "black",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 3,
        "flexDirection": "column",
        "margin": 1,
      }
    }
  />
</View>
`;

exports[`InfoBox (with title) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00ff00",
      "borderColor": "#00ff00",
      "borderRadius": 0,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "column",
      "height": 100,
      "margin": 5,
      "marginBottom": 15,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={undefined}
      style={
        Object {
          "backgroundColor": "black",
          "borderRadius": 0,
          "flex": 1,
          "height": 21,
          "margin": 2,
          "marginBottom": 1,
          "width": 10,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "flex": 9,
          "fontFamily": "px_sans_nouveaux",
          "paddingRight": 30,
          "textAlign": "center",
        }
      }
    >
      foo
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "borderColor": "black",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 3,
        "flexDirection": "column",
        "margin": 1,
      }
    }
  />
</View>
`;
